### ISSUE TITLE
Mantis Mouse Pro Beta app does not recognize keyboard input

### ISSUE THREAD

**Issue by violetclearspark** [Genymobile/scrcpy#3424]
- [✓] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [✓] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: Linux Mint 21 Vanessa Cinnamon
 - scrcpy version: 1.24
 - installation method: manual build
 - device model: motorola one fusion+
 - Android version: 10

Bug:
Mantis Mouse Pro Beta (play store link: https://play.google.com/store/apps/details?id=app.mantispro.mousekeyboard) does not recognize any keyboard input. It does recognize Middle Mouse Button, the default key to enter shooting mode, and mouse movement works as expected. A similar android app, Panda Mouse Pro (https://play.google.com/store/apps/details?id=com.panda.mouse), is able to recognize the keyboard inputs and functions as expected.

I tried checking without `-K` option, but it did not work in both the aforementioned apps. `--otg` mode works with Panda Mouse Pro, but not with Mantis Mouse Pro.

**Comment by rom1v** [comment]
> I tried checking without -K option, but it did not work in both the aforementioned apps.

For keyboard and mouse, you need `-KM`. These options are basically the same as `--otg`, but with mirroring enabled.

For your issues, I'm afraid you should report the bug to these apps :/

**Comment by violetclearspark** [comment]
No, scrcpy works well with Panda Mouse Pro, with `-KM` or `--otg` options. There's no problem with Panda Mouse Pro, I just used it as a reference.
But as for Mantis Mouse Pro, I've used an otg keyboard with the same app and it worked but it does not work with either `-KM` or `--otg` option in scrcpy.
As I mentioned, Mantis Mouse Pro app works as expected when an otg (or Bluetooth?) keyboard is used but does not respond to any keyboard event sent by scrcpy.

**Comment by rom1v** [comment]
I have no idea, unfortunately. :/

**Comment by violetclearspark** [comment]
ok, np. Also, is there a keymapping function like in Qtscrcy or do you plan to implement one in the future?

**Comment by rom1v** [comment]
See #712

**Comment by violetclearspark** [comment]
@chisato2309 I was not able to get middle mouse button or WASD working, only mouse movement. Which game were you running? Some games support WASD natively. I found no solutions, though, unfortunately.
